react/features/filmstrip/subscriber.web.js

```diff
@@ -153,3 +158,96 @@ StateListenerRegistry.register(
             store.dispatch(setTileViewDimensions(gridDimensions));
         }
     });
+
+/**
+ * Listens for changes to the remote participant count to recompute the reordered list of the remote participants.
+ */
+StateListenerRegistry.register(
+    /* selector */ getRemoteParticipantCount,
+    /* listener */ (count, store) => _updateRemoteParticipants(store));
+
+/**
+ * Listens for changes to the screensharing status of the remote participants to recompute the reordered list of the
+ * remote endpoints.
+ */
+StateListenerRegistry.register(
+    /* selector */ state => state['features/video-layout'].remoteScreenShares,
+    /* listener */ (remoteScreenShares, store) => _updateRemoteParticipants(store));
+
+StateListenerRegistry.register(
+    /* selector */ state => state['features/base/participants'].dominantSpeaker,
+    /* listener */ (dominantSpeaker, store) => _updateRemoteParticipants(store));
+
+/**
+ * Private helper to calculate the reordered list of remote participants.
+ *
+ * @param {*} store - The redux store.
+ * @returns {void}
+ */
+function _updateRemoteParticipants(store) {
+    const state = store.getState();
+    const { dominantSpeaker, local, previousSpeakers, remote } = state['features/base/participants'];
+
+    if (!remote) {
+        return;
+    }
+    const { remoteParticipants, visibleParticipants } = state['features/filmstrip'];
+    const currentParticipants = Array.from(remote.values());
+    const { remoteScreenShares } = state['features/video-layout'];
+
+    // Make a copy so that we don't modify the values in redux.
+    const screenShares = (remoteScreenShares || []).slice();
+    const compareDisplayNames = function(a, b) {
```

Comment:
Maybe you can define this function outside `_updateRemoteParticipants`?

I think when you define it here it will be recreated every time `_updateRemoteParticipants` is called. I don't think it will be a big deal but since it is not using anything from `_updateRemoteParticipants` we can easily extract it.

Reply:
Yeah, makes sense, will do that.